libcontainer/container_linux.go

```diff
@@ -388,11 +388,18 @@ func (c *Container) Signal(s os.Signal) error {
 	// leftover processes. Handle this special case here.
 	if s == unix.SIGKILL && !c.config.Namespaces.IsPrivate(configs.NEWPID) {
 		if err := signalAllProcesses(c.cgroupManager, unix.SIGKILL); err != nil {
+			if c.config.RootlessCgroups { // may not have an access to cgroup
+				return c.signal(s)
```

Comment:
> signalAllProcesses has a cgroupv1-friendly fallback

No when lacking an access to the cgroup.
It just returns `ErrNotRunning` immediately :
https://github.com/opencontainers/runc/blob/961b8031f6614be300d3b10eec9d2f54323026d9/libcontainer/init_linux.go#L696-L701

Reply:
And it seems correct that `m.Exists()` returns false from the perspective of `m`, as the cgroup actually does not exist in this case

Reply:
Right, but the function name implies that it's cgroup-specific and the fallback isn't technically cgroup-specific (though it does use the cgroup for freezing). Idk...

Reply:
signalAllProcesses is only called when shared pid ns is used (as otherwise it's enough to kill pid 1), and to know all the PIDs we need a dedicated cgroup. Meaning, that rootless container with shared pidns won't work, as there is no way to find all PIDs (well, theoretically, we can find all the children of container init in this case, but I  am not sure it's a good idea).

Not sure how it works in runc 1.1 -- most probably it just doesn't (i.e. we leak processes).